receiver/prometheusreceiver/internal/prom_to_otlp.go

```diff
@@ -54,7 +56,49 @@ func CreateNodeAndResourcePdata(job, instance, scheme string) *pcommon.Resource
 	}
 	attrs.UpsertString(conventions.AttributeServiceInstanceID, instance)
 	attrs.UpsertString(conventions.AttributeNetHostPort, port)
-	attrs.UpsertString(conventions.AttributeHTTPScheme, scheme)
+	attrs.UpsertString(conventions.AttributeHTTPScheme, serviceDiscoveryLabels.Get(model.SchemeLabel))
+
+	addKubernetesResource(attrs, serviceDiscoveryLabels)
 
 	return &resource
 }
+
+func addKubernetesResource(attrs pcommon.Map, serviceDiscoveryLabels labels.Labels) {
+	if podName := serviceDiscoveryLabels.Get("__meta_kubernetes_pod_name"); podName != "" {
+		attrs.UpsertString(conventions.AttributeK8SPodName, podName)
+	}
+	if podUID := serviceDiscoveryLabels.Get("__meta_kubernetes_pod_uid"); podUID != "" {
+		attrs.UpsertString(conventions.AttributeK8SPodUID, podUID)
+	}
+	if containerName := serviceDiscoveryLabels.Get("__meta_kubernetes_pod_container_name"); containerName != "" {
+		attrs.UpsertString(conventions.AttributeK8SContainerName, containerName)
+	}
+	if nodeName := serviceDiscoveryLabels.Get("__meta_kubernetes_pod_node_name"); nodeName != "" {
+		attrs.UpsertString(conventions.AttributeK8SNodeName, nodeName)
+	}
+	if nodeName := serviceDiscoveryLabels.Get("__meta_kubernetes_node_name"); nodeName != "" {
+		attrs.UpsertString(conventions.AttributeK8SNodeName, nodeName)
+	}
+	if nodeName := serviceDiscoveryLabels.Get("__meta_kubernetes_endpoint_node_name"); nodeName != "" {
+		attrs.UpsertString(conventions.AttributeK8SNodeName, nodeName)
+	}
+	controllerName := serviceDiscoveryLabels.Get("__meta_kubernetes_pod_controller_name")
+	controllerKind := serviceDiscoveryLabels.Get("__meta_kubernetes_pod_controller_kind")
+	if controllerKind != "" && controllerName != "" {
+		switch controllerKind {
+		case "ReplicaSet":
+			attrs.UpsertString(conventions.AttributeK8SReplicaSetName, controllerName)
+		case "DaemonSet":
+			attrs.UpsertString(conventions.AttributeK8SDaemonSetName, controllerName)
+		case "StatefulSet":
+			attrs.UpsertString(conventions.AttributeK8SStatefulSetName, controllerName)
+		case "Job":
+			attrs.UpsertString(conventions.AttributeK8SJobName, controllerName)
+		case "CronJob":
+			attrs.UpsertString(conventions.AttributeK8SCronJobName, controllerName)
+		}
+	}
```

Comment:
same here, i think we could just map from the prometheus string to the otel convention

Reply:
done